vault/resource_generic_secret.go

```diff
@@ -145,10 +147,20 @@ func genericSecretResourceWrite(d *schema.ResourceData, meta interface{}) error
 
 	}
 
-	log.Printf("[DEBUG] Writing generic Vault secret to %s", path)
-	_, err = client.Logical().Write(path, data)
-	if err != nil {
-		return fmt.Errorf("error writing to Vault: %s", err)
+	writeData := func() error {
+		if _, err := client.Logical().Write(path, data); err != nil {
+			return err
```

Comment:
This will retry on any error, which may cause unnecessary delays or perhaps introduce undesirable side-effects. It would be best to only retry on certain errors.

See https://github.com/hashicorp/terraform-provider-vault/blob/eae35c228b08cdd6636864535efcf2c5e2778d0e/vault/resource_namespace.go#L84 for a proper example of this.

Reply:
Thanks! Added this error validation.